src/lib/cms/server.ts

```diff
@@ -0,0 +1,42 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as mkdirp from 'mkdirp';
+import * as Immutable from 'immutable';
+import {IDataToUpdate, IRequestResponse,
+        IRequestResponseStatus} from './models';
+import {MongoClient, Db} from 'mongodb';
+
+const UI_STRINGS_COLLECTION: string = 'ui_strings';
+
+export function getCmsStrings(db: Db): Promise<any> {
+  return new Promise((resolve, reject) => {
```

Comment:
is there a reason you are creating a new promise ontop of the one you have already? Couldn't you just return the one you got from findLatestPublishedCmsStrings and add a .then handler to it?